staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go

```diff
@@ -93,6 +138,7 @@ func WithMaxInFlightLimit(
 			select {
 			case c <- true:
 				defer func() { <-c }()
+				watermark.record(len(nonMutatingChan), len(mutatingChan))
```

Comment:
We really want to add a blocking lock here?

Reply:
Fair point. I update a PR. Take a look if you like it better.